py-polars/polars/dataframe/frame.py

```diff
@@ -1116,6 +1118,50 @@ def _replace(self, column: str, new_column: Series) -> Self:
         self._df.replace(column, new_column._s)
         return self
 
+    @property
+    def plot(self) -> Any:
+        """
+        Create a plot namespace.
+
+        Polars does not implement plotting logic itself, but instead defers to
+        hvplot. Please see the `hvplot reference gallery <https://hvplot.holoviz.org/reference/index.html>`_
+        for more information and documentation.
+
+        Examples
+        --------
+        Scatter plot:
+
+        >>> df = pl.DataFrame(
+        ...     {
+        ...         "length": [1, 4, 6],
+        ...         "width": [4, 5, 6],
+        ...         "species": ["setosa", "setosa", "versicolor"],
+        ...     }
+        ... )
+        >>> df.plot.scatter(x="length", y="width", by="species")  # doctest: +SKIP
+
+        Line plot:
+
+        >>> from datetime import date
+        >>> df = pl.DataFrame(
+        ...     {
+        ...         "date": [date(2020, 1, 2), date(2020, 1, 3), date(2020, 1, 3)],
+        ...         "stock_1": [1, 4, 6],
+        ...         "stock_2": [1, 5, 2],
+        ...     }
+        ... )
+        >>> df.plot.line(x="date", y=["stock_1", "stock_2"])  # doctest: +SKIP
+
+        For more info on what you can pass, you can use ``hvplot.help``:
+
+        >>> import hvplot  # doctest: +SKIP
+        >>> hvplot.help("scatter")  # doctest: +SKIP
+        """
+        if not getattr(holoviews.extension, "_loaded", False):
+            # If no extension has been loaded, load 'bokeh' (the hvplot default)
+            hvplot.extension("bokeh")
+        return hvplot.plotting.core.hvPlotTabularPolars(self)
```

Comment:
@Hoxbro @maximlt I've tried writing this so it doesn't register the namespace by default

Does this look right to you?

Can Polars trust that this API will remain stable? If not, is there anything else you'd suggest using instead?

Reply:
In conjunction with https://github.com/holoviz/hvplot/pull/1243 , I think this should ensure stability

Reply:
I would suggest using [`hvplot.post_patch`](https://github.com/holoviz/hvplot/blob/734c0cc922152ea6a0f3332bc2bf19f5430d7629/hvplot/__init__.py#L164-L166) which does not show the ugly internals. 

This will then (almost) mimic what we do with `import hvplot.polars` (other than registering the `hvplot` accessor).

> In conjunction with holoviz/hvplot#1243 , I think this should ensure stability

i was thinking the same thing, but you beat me to it :+1: